web-ui/src/main/java/org/archcnl/ui/input/ListEntry.java

```diff
@@ -0,0 +1,57 @@
+package org.archcnl.ui.input;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.archcnl.domain.input.datatypes.mappings.Concept;
+import org.archcnl.domain.input.datatypes.mappings.Relation;
+
+public class ListEntry<T> {
+
+    private List<T> children;
+    private String name;
+    private T content;
+    private boolean isLeaf;
+
+    public ListEntry(String name, List<T> children) {
+        this.children = children;
+        this.name = name;
+        isLeaf = false;
+    }
+
+    public ListEntry(String name, T content) {
+        this.name = name;
+        this.content = content;
+        children = Collections.<T>emptyList();
+        isLeaf = true;
+    }
+
+    public T getContent() {
+        return content;
+    }
+
+    public List<ListEntry<T>> getHierarchicalChildren() {
+        List<ListEntry<T>> hierarchicalChildren =
+                children.stream()
+                        .map(entry -> (ListEntry<T>) new ListEntry<>(entry.toString(), entry))
+                        .collect(Collectors.toList());
+        return hierarchicalChildren;
+    }
+
+    // temporal solution, ideally Concept and Relation should have these methods as the toString
+    // method
+    @Override
+    public String toString() {
+        if (getContent() instanceof Concept) {
+            return ((Concept) getContent()).toStringRepresentation();
```

Comment:
The getName() method should be used instead of toStringRepresentation(). toStringRepresentation() is intended for the ADOC export. The user does not need to see the prefixes like "famix:" and "architecture:" as they only add unnecessary complexity while having functional consequences in the UI.
The getName() method exists for both Concept and Relation.

Reply:
Is there a reason getName() is explicitly implemented instead of overwriting toString()? toString() is used by some other classes like the TreeGrid to generate the textual representation and by overwriting it, the above code would not be necessary.

Reply:
I went with `getName()` as it allows me to force inheriting classes to overwrite it. Additionally, the name describes the functionality better than `toString()`. We could either overwrite toString()'s functionality with the getName() method in Concept and Relation or get rid of getName() altogether.